include/gpio.h

```diff
@@ -189,6 +189,20 @@ extern "C" {
  */
 #define GPIO_DS_DISCONNECT_HIGH (0x3 << GPIO_DS_HIGH_POS)
 
+/** @cond INTERNAL_HIDDEN */
+#define GPIO_DMA_REQ_POS		15
+#define GPIO_DMA_REQ_MASK		(0x3 << GPIO_DMA_REQ_POS)
+/** @endcond */
+
+/** GPIO pin trigger DMA requenst on falling edge. */
+#define GPIO_DMA_REQ_EDGE_LOW		(1 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising edge. */
+#define GPIO_DMA_REQ_EDGE_HIGH		(2 << GPIO_DMA_REQ_POS)
+
+/** GPIO pin trigger DMA requenst on rising and falling edge. */
+#define GPIO_DMA_REQ_EDGE_DOUBLE	(3 << GPIO_DMA_REQ_POS)
+
```

Comment:
I know this possibility to trigger DMA request directly from the GPIO, only with Kinetis MCUs. The other make it a little cumbersome over a timer configured for Input Capture Mode, e.g. STM32 http://www.st.com/resource/en/application_note/dm00169730.pdf

I do not know all the architectures, but if it is only possible with Kinetis MCUs, then it makes no sense to include it in the API. @MaureenHelm what do you think?